internal/instruments.go

```diff
@@ -0,0 +1,42 @@
+package internal
+
+import (
+	"go.opentelemetry.io/otel/api/global"
+	"go.opentelemetry.io/otel/api/metric"
+	"go.opentelemetry.io/otel/api/unit"
+	"log"
+)
+
+type openTelemetryInstrumentation struct {
+	// Count of write commands performed
+	WriteCount metric.Int64Counter
+	// Count of new connections
+	NewConnectionsCount metric.Int64Counter
+}
+
+var Instruments = initInstruments()
+
+func initInstruments() *openTelemetryInstrumentation {
+	meter := global.Meter("github.com/go-redis/redis")
+
+	writeCount, err := meter.NewInt64Counter("redis.writes",
+		metric.WithDescription("the number of writes initiated"),
+	)
+	if err != nil {
+		// TODO: handle errors
```

Comment:
```suggestion
		writeCount, _ = metric.NoopProvider{}.Meter("no-op").NewInt64Counter(
		        "github.com/go-redis/redis",
		)
```

I wanted to ask - I was thinking of doing this as a fallback in the event that we encounter an error. Is this a valid solution @MrAlias ?

Reply:
The default in the case of an error is to pretty much do this: https://github.com/open-telemetry/opentelemetry-go/blob/master/api/metric/sync.go#L130-L135

There isn't really much need to explicitly do this.

Reply:
The important thing is that the error is correctly bubbled up. The instrumentation should not halt normal operation of the code, but the user should have the ability to discover an error has occurred and there are no metric events being recorded.

Reply:
Oh nice I see that now 👍